packages/valory/connections/abci/connection.py

```diff
@@ -1236,18 +1236,21 @@ def check_server_status(
             try:
                 if self._monitoring.stopped():  # type: ignore
                     break  # break from the loop immediately.
-                line = self._process.stdout.readline()  # type: ignore
-                self.write_line(line)
-                for trigger in [
-                    "RPC HTTP server stopped",  # this occurs when we lose connection from the tm side
-                    "Stopping abci.socketClient for error: read message: EOF module=abci-client connection=",  # this occurs when we lose connection from the AEA side.
-                ]:
-                    if line.find(trigger) >= 0:
-                        self._stop_tm_process()
-                        self._start_tm_process()
-                        self.write_line(
-                            f"Restarted the HTTP RPC server, as a connection was dropped with message:\n\t\t {line}\n"
-                        )
+                if self._process is not None and self._process.stdout is not None:
```

Comment:
How is this suddenly an issue - makes me feel uneasy. We didn't change this recently..

Also, how do we recover from either of them being `None` with this logic?

I suspect it might be related to https://github.com/valory-xyz/open-autonomy/pull/1255 and the change this introduced to abci protocol

Reply:
I checked the GH history and this started occurring after #1218.

This specific part of the code is just monitoring so I am not sure if we should do something to recover.

Reply:
It seems the problem was introduced here: 
https://github.com/valory-xyz/open-autonomy/pull/1218/commits/48e138aaef4da999631be50f0807a29e2b1ce05a

Reply:
I will make a couple of draft PR to showcase this, and make sure that is the case and something funny is not going on my local.

Reply:
#1268 is at the last commit without the error.
#1269 is at the commit that I think introduced the problem.

Reply:
So it is not possible to run at the commit that #1268, without making changes that might effect it. So I closed it.

Regardless, it seems changing from image tag `0.1.0` -> 'latest' seemed to have caused the problem.
In #1269 there are 2 successful runs:
- One with tag "latest" and the endline [error present](https://github.com/valory-xyz/open-autonomy/runs/8155945324?check_suite_focus=true).
- One with tag "0.1.0" and the endline [error not present](https://github.com/valory-xyz/open-autonomy/runs/8155945312?check_suite_focus=true). 

Not sure what the difference between this images is. Any idea @angrybayblade ?